src/Symfony/Component/HttpFoundation/ResponseHeaderBag.php

```diff
@@ -87,12 +93,19 @@ public function all(?string $key = null): array
         if (null !== $key) {
             $key = strtr($key, self::UPPER, self::LOWER);
 
-            return 'set-cookie' !== $key ? $headers[$key] ?? [] : array_map('strval', $this->getCookies());
+            return match ($key) {
+                'set-cookie' => array_map('strval', $this->getCookies()),
+                'cache-control' => [$this->computeCacheControl()->getCacheControlHeader()],
+                default => $headers[$key] ?? [],
```

Comment:
TODO: we need to also match targeted cache control headers, not only the default one